src/Core/Filter/FrontEndObject/ConfigurationFilter.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Filter\FrontEndObject;
+
+use PrestaShop\PrestaShop\Core\Filter\HashMapWhitelistFilter;
+
+/**
+ * Filters Configuration objects that will be sent to the client
+ */
+class ConfigurationFilter extends HashMapWhitelistFilter
+{
+    public function __construct()
+    {
+        $whitelist = array(
```

Comment:
hum... looks like a configurable object... maybe a factory instead of multiples classes?

Reply:
maybe like this

```php
$factory->build($name, $whitelist) // return instance of filter interface
```
I want to extract the configuration from objects... if we do that we can see that we dont really need them, isn't it?

Reply:
> I want to extract the configuration from objects

Oh that's already the case:
```php
(new HashMapWhitelistFilter())->whitelist($whitelist)
```

My original idea was that you can mix & match generic filters in order to filter more complex structures in a more-or-less semantic way. What value would a factory add here?

Specific classes like `ProductFilter` are simply an extended filter class with a hardcoded whitelist. They serve a dual purpose:

1. Provide better semantics for mixed & nested filters
2. Encourage reusability and extensibility

I'll explain more in a bit.

Reply:
you're right, I'm frustrated to not have a computer here to play a little bit with your pull request :)